src/main/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java

```diff
@@ -0,0 +1,34 @@
+package edu.kis.powp.jobs2d.events;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import edu.kis.powp.jobs2d.command.ComplexCommand;
+import edu.kis.powp.jobs2d.command.ComplexCommandFactory;
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+import edu.kis.powp.jobs2d.magicpresets.FiguresJoe;
+
+public class SelectTestFigureOptionListener implements ActionListener {
+    private DriverManager driverManager;
+    public SelectTestFigureOptionListener(DriverManager driverManager, int i) {
+        this.driverManager = driverManager;
+    }
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        String action = e.getActionCommand();
+        if(action.equals("Figure Joe 1")){
+            FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+        }else if(action.equals("Figure Joe 2")){
+            FiguresJoe.figureScript2(driverManager.getCurrentDriver());
+        }else if(action.equals("SQUARE")){
+            ComplexCommandFactory factory = new ComplexCommandFactory(driverManager.getCurrentDriver(),0,0);
+            ComplexCommand squareFigure = factory.createRectangle(120,120);
+            squareFigure.execute();
+
+        }else if(action.equals("TRIANGLE")){
```

Comment:
Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

Reply:
NOT RESOLVED!